include/cuco/detail/open_addressing/open_addressing_ref_impl.cuh

```diff
@@ -938,7 +929,57 @@ class open_addressing_ref_impl {
   [[nodiscard]] __host__ __device__ constexpr auto const& extract_payload(
     Value const& value) const noexcept
   {
-    return thrust::get<1>(thrust::raw_reference_cast(value));
+    return thrust::raw_reference_cast(value).second;
+  }
+
+  /**
+   * @brief Converts the given type to the container's native `value_type`.
+   *
+   * @tparam T Input type which is convertible to 'value_type'
+   *
+   * @param value The input value
+   *
+   * @return The converted object
+   */
+  template <typename T>
+  [[nodiscard]] __host__ __device__ constexpr value_type native_value(T const& value) const noexcept
```

Comment:
```suggestion
  [[nodiscard]] __device__ constexpr value_type native_value(T const& value) const noexcept
```
Is host really needed?

Reply:
Not really. I tend to do this in preparation for host-sided probing on G+H but this would need much more changes. I'll remove it for now.

Reply:
Addressed in [b7ccda0](https://github.com/NVIDIA/cuCollections/pull/387/commits/b7ccda0711b951d0b9ab6f57e853810a2810f3b1)